codegen/masm/src/codegen.rs

```diff
@@ -114,15 +111,40 @@ fn quadratic_element_square(writer: &mut Writer, n: u32) {
 
 impl<'ast> CodeGenerator<'ast> {
     pub fn new(ir: &'ast AirIR, config: CodegenConfig) -> CodeGenerator<'ast> {
+        // remove duplicates and sort period lengths in descending order, since larger periods will
+        // give smaller number of cycles (which means a smaller number of exponentiations)
+        let mut periods: Vec<usize> = ir.periodic_columns().iter().map(|e| e.len()).collect();
+        periods.sort();
+        periods.dedup();
+        periods.reverse();
+
+        // map from public input name to start offset, used to compute the memory location of the
```

Comment:
minor nit: can we reformulate the sentence? I am not getting `map from public input name to start offset`

Reply:
updated

Reply:
Note: I'm going to merge this so that I can open the follow ups. Let me know if the comment still needs to be improved and I will open a follow up.